src/esp/assets/managers/AttributesManagerBase.h

```diff
@@ -998,7 +1004,7 @@ AttribsPtr AttributesManager<AttribsPtr>::createObjectAttributesFromJson(
 
 template <class T>
 bool AttributesManager<T>::setJSONAssetHandleAndType(
-    T attributes,
+    typename AttributesManager<T>::AttribsPtr attributes,
```

Comment:
Isn't `AttribsPtr` enough? :thinking:

Reply:
I'll try it.

Reply:
Apparently it is, thanks.  Need to qualify all the return types, but the argument type didn't need it.

Reply:
The return types in _out-of-class_ definitions would work only as trailing I think (`auto Class::foo() -> AttribsPtr`), the way you have them you need to prefix with class (and then with `typename`). For in-class declarations `AttribsPtr foo()` works, yes.

Reply:
I can just specify std::shared_ptr<T>, since that's what AttribsPtr is

Reply:
or that, yep, depends how you want it to appear in the docs -- if there's a typedef, i would personally use the typedef everywhere :)

Reply:
should i put them back?  seemed a bit cumbersome, but i'll differ to your judgement :)